crates/core/src/index.rs

```diff
@@ -6,18 +6,19 @@ use crate::{
     attestation::{Field, FieldId},
     transcript::{
         hash::{PlaintextHash, PlaintextHashSecret},
-        Idx,
+        Direction, Idx,
     },
 };
 
-/// Index for items which can be looked up by transcript index or field id.
+/// Index for items which can be looked up by transcript's (direction and index)
+/// or field id.
 #[derive(Debug, Clone)]
 pub(crate) struct Index<T> {
     items: Vec<T>,
     // Lookup by field id.
     field_ids: HashMap<FieldId, usize>,
-    // Lookup by transcript index.
-    transcript_idxs: HashMap<Idx, usize>,
+    // Lookup by transcript direction and index.
+    transcript_idxs: HashMap<(Direction, Idx), usize>,
```

Comment:
bug: `Direction` needs to be stored alongside `Idx` as the latter doesn't contain any direction info.

Reply:
good catch, i'll open an issue lest we forget (but i think this bug was already addressed in one of the open PRs, or maybe it's in my local wip review branch.)

Reply:
ah i actually also fixed this bug in this PR itself